### PR TITLE
Replaces Lambda@Edge with CloudFront Function

### PR DESCRIPTION
**Pull request by nvnivs**
* Refactors Lambda@Edge to CloudFront Function: https://github.com/TechToSpeech/terraform-aws-serverless-static-wordpress/issues/57
* Implements `cloudfront_function_301_redirects` variable to enable users to pass a map of custom permanent redirects

### CODE REVIEW

File: modules/cloudfront/main.tf [hunk 1]
```diff
@@ -1,76 +1,20 @@
...
-      type        = "Service"
-      identifiers = ["edgelambda.amazonaws.com", "lambda.amazonaws.com"]
+resource "aws_cloudfront_function" "object_rewrite" {
+  name    = "${var.site_name}_rewrite"
+  runtime = "cloudfront-js-1.0"
+  publish = true
+  code    = templatefile(
+    "${path.module}/function_rewrite/index.js.tftpl",
+    {
+      REDIRECTS = var.cloudfront_function_301_redirects
```
Comment (petewilcock) [inline modules/cloudfront/main.tf:21]: Could we get a `depends_on` block here on the log group in the same way the lambda function did? This should ensure the log group is created first by Terraform with the retention settings. If the function creates the log group first, TF will complain a group with that name already exists.

**Review by petewilcock — CHANGES_REQUESTED** [review]
Looks really promising! Just one note. Look forward to try this against my personal project.

File: modules/cloudfront/main.tf (hunk 1 above)
Comment (nvnivs) [reply modules/cloudfront/main.tf:21]: Fixed. Got bitten by this earlier and then still forgot to add the dependency...

**Review by petewilcock — APPROVED** [review]

**Comment by petewilcock** [comment]
Closes #57

**Comment by petewilcock** [comment]
Great stuff @z0c thank you - this is a really helpful contribution, I'm quite excited for 0.2.0!

**Pull request merged**